### ISSUE TITLE
Use EPE-NAS for custom classification

### ISSUE THREAD

**Issue by Mshz2** [VascoLopes/EPE-NAS#1]
Hi there, I really like your approach and would like to use your code for my thesis. Would you please give me a very short clue about how can I apply  EPE-NAS with a random search strategy (the Random Search and Reproducibility NAS that you used in the paper) for an custom dataset for two classes? I like to find the best classification architecture with NAS via your solution.
should I just change dataset path and dataloader? or it is in contradict with NAS-Bench-201.pth file?
I really appreciate your help.
Best

**Comment by VascoLopes** [comment]
Hi @Mshz2,

To use a custom dataset, you need to:
1) have a data loader for your problem/data set
2) use random search to sample architectures (you can use the NAS-Bench-201 search space as well)
3) train the highest scoring network from scratch, instead of getting the statistics from the NAS-Bench-201 API

The easiest way would be to just change the dataset and instead of getting the statistics for the final network, train it using your favourite training scheme.

**Comment by Mshz2** [comment]
> Hi @Mshz2,
> 
> To use a custom dataset, you need to:
> 
> 1. have a data loader for your problem/data set
> 2. use random search to sample architectures (you can use the NAS-Bench-201 search space as well)
> 3. train the highest scoring network from scratch, instead of getting the statistics from the NAS-Bench-201 API
> 
> The easiest way would be to just change the dataset and instead of getting the statistics for the final network, train it using your favourite training scheme.

Thanks a lot for the quick response. May I ask again in which part of the codes could I get or replace the final (best) network? Would it be like a binary vector or an object? 
Would be appropriate if you answer this as I'm beginner 🙈, what do you mean by training scheme in here?

Thanks a lot again

**Comment by VascoLopes** [comment]
The final network is obtained in line 204 in the file `search.py`. You should replace line 205 for `config = api.get_net_config(best_arch, args.dataset)` and then train the network as you prefer (number of epochs, optimizer, loss function, ...).

**Comment by Mshz2** [comment]
> The final network is obtained in line 204 in the file `search.py`. You should replace line 205 for `config = api.get_net_config(best_arch, args.dataset)` and then train the network as you prefer (number of epochs, optimizer, loss function, ...).

Amazing! I also just saw the API.get_net_config in line 174. In line 177 the network object for passing to training iteration is obtained by 

`        network = get_cell_based_tiny_net(config)  #create the network from configuration
        network = network.to(device)
`
,Right?

**Comment by Mshz2** [comment]
> Hi @Mshz2,
> 
> To use a custom dataset, you need to:
> 
> 1. have a data loader for your problem/data set
> 2. use random search to sample architectures (you can use the NAS-Bench-201 search space as well)
> 3. train the highest scoring network from scratch, instead of getting the statistics from the NAS-Bench-201 API
> 
> The easiest way would be to just change the dataset and instead of getting the statistics for the final network, train it using your favourite training scheme.

May I ask from where can I select to use either  random search to sample architectures or the NAS-Bench-201 search space? seems like search.py uses --api_loc './datasets/NAS-Bench-201-v1_0-e61699.pth' via args. Do I need to type sth else in order to apply random search api? Many thanks

**Comment by VascoLopes** [comment]
Random search is the processing that refers to the sampling of networks by random. In our case, random search is used to randomly sample networks from the NAS-Bench-201 search space (which defines the architectures that can be created).
